File: backend/app/schemas/gamification.py
"""
LAYERS - Badge & Leaderboard Schemas
"""

from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel


# ---- Badges ----

class BadgeItem(BaseModel):
    id: str
    title: str
    description: str
    icon: str
    unlocked: bool
    unlocked_at: Optional[datetime] = None


class BadgesResponse(BaseModel):
    badges: List[BadgeItem]
    unlocked_count: int
    total: int


class BadgeUnlockedItem(BaseModel):
    id: str
    title: str
    icon: str


# ---- Leaderboard ----

class LeaderboardEntry(BaseModel):
    rank: int
    user_id: str
    username: str
    avatar_url: Optional[str] = None
    score: int
    is_me: bool = False


class LeaderboardResponse(BaseModel):
    scope: str
    entries: List[LeaderboardEntry]
    my_rank: Optional[int] = None
    my_score: int = 0

File: mobile/src/types/gamification.ts
/**
 * LAYERS — Gamification Types (Week 7 Day 5)
 */

// ---- Badges ----
export interface BadgeItem {
  id: string;
  title: string;
  description: string;
  icon: string;
  unlocked: boolean;
  unlocked_at?: string | null;
}

export interface BadgesResponse {
  badges: BadgeItem[];
  unlocked_count: number;
  total: number;
}

// ---- Leaderboard ----
export type LeaderboardScope = "global" | "weekly";

export interface LeaderboardEntry {
  rank: number;
  user_id: string;
  username: string;
  avatar_url?: string | null;
  score: number;
  is_me: boolean;
}

export interface LeaderboardResponse {
  scope: string;
  entries: LeaderboardEntry[];
  my_rank?: number | null;
  my_score: number;
}
